app/org/sagebionetworks/bridge/services/ScheduledActivityService.java

```diff
@@ -214,6 +226,28 @@ public void deleteActivitiesForUser(String healthCode) {
             .collect(toImmutableList());
     }
     
+    /**
+     * If the client data is being added or removed, or if it is different, then the activity is being 
+     * updated.
+     */
+    protected boolean hasUpdatedClientData(ScheduledActivity schActivity, ScheduledActivity dbActivity) {
+        JsonNode schNode = (schActivity == null) ? null : schActivity.getClientData();
+        JsonNode dbNode = (dbActivity == null) ? null : dbActivity.getClientData();
+        
+        // One is null and the other is not, or they are not equal (JsonNode equals() accepts 
+        // null parameter, so it is testing both).
+        return (schNode == null && dbNode != null) ||
+               (schNode != null && !schNode.equals(dbNode));
```

Comment:
To clarify, I meant https://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#equals(java.lang.Object,%20java.lang.Object), which I'm 99% sure would both be simpler and more readable.